msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/DeviceCodeFlowApiTest.java

```diff
@@ -0,0 +1,329 @@
+        config.setAuthorityUrl(new URL(mUrlBody));
+        final OAuth2StrategyParameters options = new OAuth2StrategyParameters();
+        final OAuth2Strategy strategy = new MicrosoftStsOAuth2Strategy(config, options);
+
+        // Previously authenticated code
+        mTokenRequest.setDeviceCode(
+                "AAQABAAEAAAAm-06blBE1TpVMil8KPQ41e5vDLI7te0y-3XHYO_uurPryAiyBiPiKnjEVzAQZQzCyGZERne4a" +
+                        "IwYAiBlQ8an93ENYuVOO-vEAt48FEJSEMQqq-zHZVD59bkc6eYIAViZKVvTv5_qilKj4uEjVE9BGkIxY5B6Uq1K8oWHEqzH-w6CiWjC8vQc6mSV_FPCbnAggAA");
+
+        final TokenResult tokenResult = strategy.requestToken(mTokenRequest);
+        Assert.assertNull(tokenResult.getTokenResponse());
+        Assert.assertNotNull(tokenResult.getErrorResponse());
+        Assert.assertEquals(ErrorStrings.DEVICE_CODE_FLOW_EXPIRED_TOKEN_CODE, tokenResult.getErrorResponse().getError());
+    }
+
+    // A device code that has not yet been registered leads to invalid_grant, not bad_verification_code
+//    @Test
+//    public void testDeviceCodeFlowTokenBadVerificationCode() throws IOException, ClientException {
+//        final MicrosoftStsOAuth2Configuration config = new MicrosoftStsOAuth2Configuration();
+//        config.setAuthorityUrl(new URL(mUrlBody));
+//        final OAuth2StrategyParameters options = new OAuth2StrategyParameters();
+//        final OAuth2Strategy strategy = new MicrosoftStsOAuth2Strategy(config, options);
+//        mTokenRequest.setDeviceCode(
+//                "AAQABAAEAAAAm-06blBE1TpVMil8KPQ41e5vDLI7te0y-3XHYO_uurPryAiyBiPiKnjEVzAQZQzCyGZERne4a" +
+//                        "IwYAiBlQ8an93ENYuVOO-vEAt48FEJSEMQqq-zHZVD59bkc6eYIAViZKVvTv5_qilKj4uEjVE9BGkIxY5B6Uq1K8oWHEqzH-w6CiWjC8vQc6mSV_FPCbnAggBA");
+//
+//        final TokenResult tokenResult = strategy.requestToken(mTokenRequest);
+//        Assert.assertNull(tokenResult.getTokenResponse());
+//        Assert.assertNotNull(tokenResult.getErrorResponse());
+//        Assert.assertEquals(ErrorStrings.DEVICE_CODE_FLOW_BAD_VERIFICATION_CODE, tokenResult.getErrorResponse().getError());
+//    }
+
+    // authorization_declined is triggered in the actual auth side
+//    @Test
+//    public void testDeviceCodeFlowTokenAuthorizationDeclined() throws IOException, ClientException {
+//        final MicrosoftStsOAuth2Configuration config = new MicrosoftStsOAuth2Configuration();
+//        config.setAuthorityUrl(new URL(mUrlBody));
+//        final OAuth2StrategyParameters options = new OAuth2StrategyParameters();
+//        final OAuth2Strategy strategy = new MicrosoftStsOAuth2Strategy(config, options);
+//
+//        final TokenResult tokenResult = strategy.requestToken(mTokenRequest);
+//        Assert.assertNull(tokenResult.getTokenResponse());
+//        Assert.assertNotNull(tokenResult.getErrorResponse());
+//        Assert.assertEquals(ErrorStrings.DEVICE_CODE_FLOW_AUTHORIZATION_DECLINED_CODE, tokenResult.getErrorResponse().getError());
+//    }
```

Comment:
Should these be removed?

Reply:
I think so, I wanted to let people take a look at them before I removed them, but I don't think they serve much purpose as is.